presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PropertyDerivations.java

```diff
@@ -183,13 +183,23 @@ public ActualProperties visitEnforceSingleRow(EnforceSingleRowNode node, List<Ac
         public ActualProperties visitAssignUniqueId(AssignUniqueId node, List<ActualProperties> inputProperties)
         {
             ActualProperties properties = Iterables.getOnlyElement(inputProperties);
+
+            ImmutableList.Builder<LocalProperty<Symbol>> newLocalProperties = ImmutableList.builder();
+            newLocalProperties.addAll(properties.getLocalProperties());
```

Comment:
Probably you should strip constant leading properties from `properties.getLocalProperties()` because we are handling those explicitly in `ActualProperties.Builder#constants`.

In fact I would rename `ActualProperties.Builder#constants` to more explicit `leadingConstants`. Where `ActualProperties#getConstants` is used anyway. Should we bother with those `NullableValues`?